draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -445,9 +450,23 @@ Algorithms in this document make use of utility functions described below.
   formula directly, and annotate the usage with this definition.
 
 - sqrt(x, p):
-    Computing square roots should be done in constant time where possible.
-    If p = 3 (mod 4): sqrt(x, p) := x^(p+1)/4
-    Else: (TODO) use a suitable constant-time Tonelli-Shanks variant
+  Computing square roots should be done in constant time where possible.
+
+  When p = 3 (mod 4), the square root can be computed as `sqrt(x, p) := x^(p+1)/4`.
+  This is true for P256, P384, and Curve448.
+
+  When p = 5 (mod 8), the square root can be computed by the following algorithm.
+  This is true for Curve25519.
+
+~~~
+  sqrt(x, p) :=
+                 x^(p+3)/8     if x^(p+3)/4 == x
+      sqrt(-1) * x^(p+3)/8     otherwise
```

Comment:
Should we add a comment which says this is -1 mod p? (I'm assuming it is, so please correct me and help clarify if I'm wrong!)

Reply:
It is indeed, and I agree it is worth clarifying (especially since it should be precomputed)